pkg/cloud/services/autoscaling/autoscalinggroup.go

```diff
@@ -126,34 +126,17 @@ func (s *Service) ASGIfExists(name *string) (*expinfrav1.AutoScalingGroup, error
 	case err != nil:
 		record.Eventf(s.scope.InfraCluster(), "FailedDescribeAutoScalingGroups", "failed to describe ASG %q: %v", *name, err)
 		return nil, errors.Wrapf(err, "failed to describe AutoScaling Group: %q", *name)
+	case len(out.AutoScalingGroups) == 0:
```

Comment:
We call this code from 2 places:
1. https://github.com/kubernetes-sigs/cluster-api-provider-aws//blob/6ff9319973b929cb4a2ffceca37bbb748e048c44/exp/controllers/awsmachinepool_controller.go#L342. If we do not find an ASG, we record an Event here: https://github.com/kubernetes-sigs/cluster-api-provider-aws//blob/6ff9319973b929cb4a2ffceca37bbb748e048c44/exp/controllers/awsmachinepool_controller.go#L349.
2. https://github.com/kubernetes-sigs/cluster-api-provider-aws//blob/6ff9319973b929cb4a2ffceca37bbb748e048c44/exp/controllers/awsmachinepool_controller.go#L263. if we do not find an ASG, we do not record any Event. I suggest recording an Event here. WDYT?



(As an aside, using the `FailedDescribeAutoScalingGroups` event seems incorrect. We successfully described the autoscaling groups. We just did not find a group matching our name.)

Reply:
Good catch, I'll take a closer look at this.

Reply:
Thank you!

Reply:
I switched the `FailedDescribeAutoScalingGroups` to a `corev1.EventTypeNormal` and added an events to the `reconcileNormal` loop. There are two places added, one if the asg does not exist in `reconcileNormal`, and a second event emitted if the instance refresh is enabled as well. Not sure if we want to distinguish them in some way.

Reply:
Thanks!

What do you think if, as part of this PR, we perform a small clean-up: For consistency, let's replace the string literal `"NoASGFound"` with `expinfrav1.ASGNotFoundReason` everywhere.

Reply:
Should all be addressed now.